internal/provider/resource_pool.go

```diff
@@ -133,13 +167,13 @@ func resourcePoolCreate(ctx context.Context, d *schema.ResourceData, meta interf
 		}
 	}
 
-	vdev_spec, err := parseVdevSpecification(d.Get("mirror"), d.Get("devices"))
```

Comment:
This isn't really related, but looks like creating a pool using `device` blocks was broken, and I was trying to use that for testing.

Reply:
Definitely an oversight, good catch. The terraform project I was using as a test only used mirrors, so I never hit that issue.